

# Changeset Information

Local author: Sean Champ <spchamp@users.noreply.github.com>

Changeset range: f992d09e25d [upstream] to 11b8df3f6ff [local]
Changeset filter @ pkgsrc: editors/emacs editors/emacs26, local authorship

Upstream parent changeset @ pkgsrc: f992d09e25d
Local parent changeset @ thinkum pkgsrc contrib: 8932e6eaf30

Corresponds to a patch on pkgsrc main ports tree:
patch_wip-emacs-support.diff


diff --git a/editors/emacs/modules.mk b/editors/emacs/modules.mk
index 77faab3..899dcf5 100644
--- a/editors/emacs/modules.mk
+++ b/editors/emacs/modules.mk
@@ -1,4 +1,4 @@
-# $NetBSD: modules.mk,v 1.26 2019/06/08 10:40:54 rillig Exp $
+# $NetBSD$
 #
 # This Makefile fragment handles Emacs Lisp Packages (== ELPs).
 #
@@ -228,7 +228,7 @@ BUILD_DEFS_EFFECTS+=	${_SYS_VARS.emacs}
 
 _EMACS_VERSIONS_ALL= \
 	emacs20 emacs21 emacs21nox emacs25 emacs25nox emacs26 emacs26nox\
-	xemacs214 xemacs214nox xemacs215 xemacs215nox
+	emacs27 emacs27nox xemacs214 xemacs214nox xemacs215 xemacs215nox
 
 _EMACS_PKGDIR_MAP= \
 	emacs20@../../editors/emacs20 \
@@ -238,6 +238,8 @@ _EMACS_PKGDIR_MAP= \
 	emacs25nox@../../editors/emacs25-nox11 \
 	emacs26@../../editors/emacs26 \
 	emacs26nox@../../editors/emacs26-nox11 \
+	emacs27@../../wip/emacs-git \
+	emacs27nox@../../wip/emacs-nox11-git \
 	xemacs214@../../editors/xemacs \
 	xemacs214nox@../../editors/xemacs-nox11 \
 	xemacs215@../../editors/xemacs-current \
@@ -349,7 +351,7 @@ PRINT_PLIST_AWK+=	{ gsub(/${EMACS_LISPPREFIX:S|${PREFIX}/||:S|/|\\/|g}/, \
 .if defined(EMACS_BUILDLINK)
 _EMACS_DIR=	${BUILDLINK_DIR}/share/emacs
 #  development version usually claims three digit, say, 27.0.50 etc
-.  if ${EMACS_TYPE} == emacs27
+.  if !empty(EMACS_TYPE:Memacs27*)
 ALL_ENV+=	EMACSLOADPATH=${_EMACS_DIR}/${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}.${_EMACS_VERSION_MICRO}/lisp:${_EMACS_DIR}/site-lisp
 .  else
 ALL_ENV+=	EMACSLOADPATH=${_EMACS_DIR}/${_EMACS_VERSION_MAJOR}.${_EMACS_VERSION_MINOR}/lisp:${_EMACS_DIR}/site-lisp
diff --git a/editors/emacs26/Makefile.common b/editors/emacs26/Makefile.common
index 60af92f..d30a22d 100644
--- a/editors/emacs26/Makefile.common
+++ b/editors/emacs26/Makefile.common
@@ -2,14 +2,18 @@
 #
 # used by editors/emacs26/Makefile
 # used by editors/emacs26-nox11/Makefile
+# used by wip/emacs-git/Makefile.common
 
-.include "../../mk/bsd.prefs.mk"
+.if !defined(EMACS_COMMON_MK)
+EMACS_COMMON_MK=	#defined
 
 DISTNAME=	emacs-${EMACS_VERSION}
 PKGNAME?=	${DISTNAME:S/emacs/emacs26/}
 CATEGORIES=	editors
 MASTER_SITES=	${MASTER_SITE_GNU:=emacs/}
 
+.include "../../mk/bsd.prefs.mk"
+
 MAINTAINER=	pkgsrc-users@NetBSD.org
 HOMEPAGE=	https://www.gnu.org/software/emacs/
 COMMENT=	GNU editing macros (editor)
@@ -72,8 +76,12 @@ CPPFLAGS+=	-DTERMINFO
 
 .include "../../mk/oss.buildlink3.mk"
 
+.if !defined(EMACS_GIT_COMMON_MK)
 post-extract:
 	cp ${.CURDIR}/../../editors/emacs26/files/site-init.el ${WRKSRC}/lisp
 	${CHMOD} -R go-w ${WRKSRC}
+.endif
 
 .include "../../devel/libgetopt/buildlink3.mk"
+
+.endif ## EMACS_COMMON_MK
